File: ruby/backend-validator/03-multiple-countries.rb
require 'bundler/setup'
require 'countriesdb-validator'
require 'json'

# Set your CountriesDB private API key here
api_key = 'YOUR_API_KEY' # Replace with your actual API key
if api_key == 'YOUR_API_KEY'
  abort 'Please set your API_KEY in the script'
end

validator = CountriesDB::Validator.new(api_key: api_key)

# 3. Multiple country validation
codes = ['US', 'CA', 'MX', 'FR']
puts "Validating country codes: #{codes.join(', ')}"
results = validator.validate_countries(codes)

puts "Response: #{JSON.pretty_generate(results)}"

results.each do |result|
  unless result[:valid]
    abort "❌ Country #{result[:code]} validation failed: #{result[:message]}"
  end
end

puts '✅ All countries are valid'

File: go/backend-validator/03-multiple-countries.go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"log"

	"github.com/countriesdb/validator-go"
)

func main() {
	// Set your CountriesDB private API key here
	apiKey := "YOUR_API_KEY" // Replace with your actual API key
	if apiKey == "YOUR_API_KEY" {
		log.Fatal("Please set your API_KEY in the script")
	}

	v, err := validator.NewValidator(apiKey)
	if err != nil {
		log.Fatal(err)
	}

	ctx := context.Background()

	// 3. Multiple country validation
	codes := []string{"US", "CA", "MX", "FR"}
	fmt.Printf("Validating country codes: %v\n", codes)
	results, err := v.ValidateCountries(ctx, codes, validator.CountryOptions{})
	if err != nil {
		log.Fatalf("Validation failed: %v", err)
	}

	jsonResult, _ := json.MarshalIndent(results, "", "  ")
	fmt.Println("Response:", string(jsonResult))

	for _, result := range results {
		if !result.Valid {
			log.Fatalf("❌ Country %s validation failed: %s", result.Code, result.Message)
		}
	}

	fmt.Println("✅ All countries are valid")
}
